pkg/k8s/watchers/watcher.go

```diff
@@ -337,9 +337,8 @@ func (*k8sMetrics) Increment(_ context.Context, code string, method string, host
 	// more info:
 	// https://github.com/kubernetes/client-go/blob/v0.18.0-rc.1/rest/request.go#L700-L703
 	if code != "<error>" {
-		// Consider success if status code is 2xx or 4xx
-		if strings.HasPrefix(code, "2") ||
-			strings.HasPrefix(code, "4") {
+		// Consider success only if status code is 2xx
+		if strings.HasPrefix(code, "2") {
```

Comment:
I'm wondering if maybe covering the two cases via labels (i.e. "backend is ok and request was ok" and "backend was ok but there was client error") would still be useful for certain scenarios?

Reply:
Definitely could be useful, but the semantics are a bit fuzzy. In our specific case, we get a 401 and want to understand that we need to reconnect, whereas for a 404 we definitely don't. I don't see how we can usefully distinguish between different 4xx  concepts here, so I opted for just erring on the cautious side.
I also don't have an actual use case for the distinction just yet, so maybe we could think about this more if one arises.